### ISSUE TITLE
Simple Tokenizer not separating punctuation correctly

### ISSUE THREAD

**Issue by juanjoDiaz** [adbar/simplemma#35]
It seems that punctuation symbols are not correctly separated.
```python
>>> simple_tokenizer('"test": "that"')
['"', 'test', '":', '"', 'that', '"']
````
I would have expected:
['"', 'test', '"', ':', '"', 'that', '"']

P.S.: I would rather to review my big refactor before applying any other changes to avoid a constant conflict in the PR 🙂

**Comment by adbar** [comment]
Thanks for the feedback!

The tokenizer does something slightly different than usually expected: it clusters chars together while segmenting the input. Since the output only consists of lemmata the idea is to keep it simple and to group punctuation signs because they're not relevant in this case.

Maybe the name could be changed (word tokenizer?), this behavior should at least be documented.

**Comment by adbar** [comment]
Yes, your PR has the priority now!

**Comment by juanjoDiaz** [comment]
Hi @adbar,

Coming back to this.
Is there any reason to group symbols? Performance or something else?
Or was this just because it was simpler and it does the job as symbols are ignored?

**Comment by adbar** [comment]
Yes, it's faster and simpler. Otherwise you would have to tokenize punctuation accurately (which is a different task) and run the lemmatizer on it (which is useless in the current context and only means additional processing time).

**Comment by adbar** [comment]
Already solved upstream, get even better with #182.
